### ISSUE TITLE
`Style/FormatString` autocorrection can lead to issues

### ISSUE THREAD

**Issue by stdedos** [rubocop/rubocop#13367]
```ruby
class BaseFormatter
  attr_accessor :format

  def initialize(format)
    @format = format
  end

  def manipulate_and_print
    puts "Base format: %s" % format
  end
end

class Formatter < BaseFormatter
  def manipulate_and_print
    manipulated = format.upcase
    puts "%s: %s" % [format, manipulated]
  end
end

# Example usage
base_formatter = BaseFormatter.new("Base Format")
base_formatter.manipulate_and_print

formatter = Formatter.new("Hello, World!")
formatter.manipulate_and_print
```

## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation.

idk. Honestly this is a close call. It seems (as expected and anticipated) that `rubocop` cannot "guess" if something else is in the scope.
Although, "annoyingly enough", said attribute (`format`) is part of the formatting itself.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

```ruby
class BaseFormatter
  attr_accessor :format

  def initialize(format)
    @format = format
  end

  def manipulate_and_print
    puts 'Base format: %s' % format
  end
end

class Formatter < BaseFormatter
  def manipulate_and_print
    manipulated = format.upcase
    puts format('%s: %s', format, manipulated)
  end
end

# Example usage
base_formatter = BaseFormatter.new('Base Format')
base_formatter.manipulate_and_print

formatter = Formatter.new('Hello, World!')
formatter.manipulate_and_print
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.67.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, analyzing as Ruby 3.3, running on ruby 3.3.5) [x86_64-linux]
  - rubocop-performance 1.22.1
  - rubocop-rspec 3.1.0
```

**Comment by dvandersluis** [comment]
Rubocop cannot detect that you have overridden `Kernel#format`, particularly in another file. I'd highly suggest not doing such an override, or use `rubocop:disable` for your particular `%` format usage.

**Comment by stdedos** [comment]
_It's not completed, it's won't fix_ 🙏

**Comment by stdedos** [comment]
pls @dvandersluis

**Comment by bbatsov** [comment]
Well, it can't really be fixed as noted above, given the static analysis limitations. I'm not sure what kind of a resolution are you expecting here.

**Comment by stdedos** [comment]
To close the issue as _won't fix_, not _completed_ 

GitHub has a distinct status for that

**Comment by bbatsov** [comment]
Done.
